catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -723,6 +758,18 @@ public SolrQuery dwithin(String propertyName, String wkt, double distance) {
         return new SolrQuery(pointRadiusQuery);
       } else {
         Geometry bufferGeo = geo.buffer(distanceInDegrees, QUADRANT_SEGMENTS);
+        final Envelope envelope = bufferGeo.getEnvelopeInternal();
+        final boolean crossesDateline =
+            envelope.getWidth() >= 180
+                || (envelope.getMinX() <= 180 && envelope.getMaxX() > 180)
+                || (envelope.getMinX() < -180 && envelope.getMaxX() >= -180);
+        // When spatial4j's JtsGeometry unwraps polygons that cross the dateline, it checks that all
+        // interior rings of the polygon are subsets of the exterior ring, which means polygons with
+        // holes will throw an exception. Remove any holes from polygons that cross the dateline to
+        // prevent that exception from being thrown.
+        if (crossesDateline) {
+          bufferGeo = removeHoles(bufferGeo);
```

Comment:
Looks like there is a generic hole polygon hole remover but it is in the jtslab package so it is not production ready yet.

https://github.com/locationtech/jts/blob/0afbfb1956ec24912a8b4dc4edff0f1200442857/modules/lab/src/main/java/org/locationtech/jtslab/clean/HoleRemover.java

Reply:
Good find; I will add a comment saying we should look into switching to this once it's ready.